### PR TITLE
Update Semaphore.xml

### PR DESCRIPTION
**Pull request by HugoRoss**
Added detailed description to the name parameters, explaining the 4 possibilities (prefix with "Global\", prefix with "Local\", name without prefix, null or empty string).

### CODE REVIEW

**Comment by mairaw** [comment]
Thank you @HugoRoss for your contribution. @kouvel can you please review this change?

**Comment by HugoRoss** [comment]
Hint: My additions only cover the Windows side. @Reviewer and the public: If on Linux semaphores use a different convention or work otherwise differently, that should also be added to the description. (But in my humble opinion .NET programmers have to know these information and one cannot expect them to invest some hours investigating low level API documentations to figure out what might be going on under the hood.)

File: xml/System.Threading/Semaphore.xml [hunk 1]
```diff
@@ -189,7 +189,7 @@
       <Docs>
         <param name="initialCount">The initial number of requests for the semaphore that can be granted concurrently.</param>
         <param name="maximumCount">The maximum number of requests for the semaphore that can be granted concurrently.</param>
-        <param name="name">The name of a named system semaphore object.</param>
+        <param name="name">The name of the semaphore. If it is <see langword="null" /> or empty, a process-local semaphore is created. If the name is prefixed by "Global\", a machine-wide semaphore is created (or the existing one returned). If it is prefixed by "Local\", a session-wide semaphore is created (or the existing one returned). If the name is not prefixed at all, a session-wide semaphore is created (same as prefix "Local\"). Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (kouvel) [inline xml/System.Threading/Semaphore.xml]: Maybe this instead:

```suggestion
        <param name="name">The name of the semaphore. If it is <see langword="null" /> or empty, a process-local semaphore is created. If the name is prefixed by "Global\", a machine-wide semaphore is created (or the existing one returned). If it is prefixed by "Local\", a session-wide semaphore is created (or the existing one returned). If the name is not prefixed at all, a session-wide semaphore is created (same as prefix "Local\"). Both prefix and name are case-sensitive.</param>
```

Similarly elsewhere.

File: xml/System.Threading/Semaphore.xml [hunk 2]
```diff
@@ -282,7 +282,7 @@
       <Docs>
         <param name="initialCount">The initial number of requests for the semaphore that can be satisfied concurrently.</param>
         <param name="maximumCount">The maximum number of requests for the semaphore that can be satisfied concurrently.</param>
-        <param name="name">The name of a named system semaphore object.</param>
+        <param name="name">The name of the semaphore. If it is <see langword="null" /> or empty, a process-local semaphore is created. If the name is prefixed by "Global\", a machine-wide semaphore is created (or the existing one returned). If it is prefixed by "Local\", a session-wide semaphore is created (or the existing one returned). If the name is not prefixed at all, a session-wide semaphore is created (same as prefix "Local\"). Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:285]:
```suggestion
        <param name="name">The name of the semaphore. If it is <see langword="null" /> or empty, a process-local semaphore is created. If the name is prefixed by "Global\", a machine-wide semaphore is created (or the existing one returned). If it is prefixed by "Local\", a session-wide semaphore is created (or the existing one returned). If the name is not prefixed at all, a session-wide semaphore is created (same as prefix "Local\"). Both prefix and name are case-sensitive.</param>
```

File: xml/System.Threading/Semaphore.xml [hunk 3]
```diff
@@ -361,7 +361,7 @@
...
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:364]:
```suggestion
        <param name="name">The name of the semaphore. If it is <see langword="null" /> or empty, a process-local semaphore is created. If the name is prefixed by "Global\", a machine-wide semaphore is created (or the existing one returned). If it is prefixed by "Local\", a session-wide semaphore is created (or the existing one returned). If the name is not prefixed at all, a session-wide semaphore is created (same as prefix "Local\"). Both prefix and name are case-sensitive.</param>
```

File: xml/System.Threading/Semaphore.xml [hunk 4]
```diff
@@ -533,7 +533,7 @@
         <Parameter Name="name" Type="System.String" />
       </Parameters>
       <Docs>
-        <param name="name">The name of the system semaphore to open.</param>
+        <param name="name">The name of the semaphore. If it is prefixed by "Global\", it refers to a machine-wide semaphore, if it is prefixed by "Local\" or does not have a prefix at all, it refers to a session-wide semaphore. Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:536]:
```suggestion
        <param name="name">The name of the semaphore. If it's prefixed by "Global\", it refers to a machine-wide semaphore. If it's prefixed by "Local\" or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
```

File: xml/System.Threading/Semaphore.xml [hunk 5]
```diff
@@ -611,7 +611,7 @@
         <Parameter Name="rights" Type="System.Security.AccessControl.SemaphoreRights" Index="1" FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
       </Parameters>
       <Docs>
-        <param name="name">The name of the system semaphore to open.</param>
+        <param name="name">The name of the semaphore. If it is prefixed by "Global\", it refers to a machine-wide semaphore, if it is prefixed by "Local\" or does not have a prefix at all, it refers to a session-wide semaphore. Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:614]:
```suggestion
        <param name="name">The name of the semaphore. If it's prefixed by "Global\", it refers to a machine-wide semaphore. If it's prefixed by "Local\" or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
```

File: xml/System.Threading/Semaphore.xml [hunk 6]
```diff
@@ -938,7 +938,7 @@
         <Parameter Name="result" Type="System.Threading.Semaphore" RefType="out" Index="1" FrameworkAlternate="dotnet-uwp-10.0;netcore-1.0;netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-1.0;netstandard-1.1;netstandard-1.2;netstandard-1.3;netstandard-1.4;netstandard-1.6;netstandard-2.0;netstandard-2.1;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
       </Parameters>
       <Docs>
-        <param name="name">The name of the system semaphore to open.</param>
+        <param name="name">The name of the semaphore to open. If it is prefixed by "Global\", it refers to a machine-wide semaphore, if it is prefixed by "Local\" or does not have a prefix at all, it refers to a session-wide semaphore. Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:941]:
```suggestion
        <param name="name">The name of the semaphore. If it's prefixed by "Global\", it refers to a machine-wide semaphore. If it's prefixed by "Local\" or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
```

File: xml/System.Threading/Semaphore.xml [hunk 7]
```diff
@@ -995,7 +995,7 @@
         <Parameter Name="result" Type="System.Threading.Semaphore" RefType="out" Index="2" FrameworkAlternate="netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
       </Parameters>
       <Docs>
-        <param name="name">The name of the system semaphore to open.</param>
+        <param name="name">The name of the semaphore to open. If it is prefixed by "Global\", it refers to a machine-wide semaphore, if it is prefixed by "Local\" or does not have a prefix at all, it refers to a session-wide semaphore. Watch out: Both prefix and name are case-sensitive!</param>
```
Comment (gewarren) [inline xml/System.Threading/Semaphore.xml:998]:
```suggestion
        <param name="name">The name of the semaphore. If it's prefixed by "Global\", it refers to a machine-wide semaphore. If it's prefixed by "Local\" or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
```

**Comment by gewarren** [comment]
Close/reopen for build.

**Pull request merged**